ddtrace/internal/agent.py

```diff
@@ -44,7 +63,12 @@ def get_trace_url():
 
     Raises a ``ValueError`` if the URL is not supported by the Agent.
     """
-    url = os.environ.get("DD_TRACE_AGENT_URL", "http://%s:%d" % (get_hostname(), get_trace_port()))
+    url = None
```

Comment:
Logic should be:

- If trace agent url is set use that
- othewise, If Host or Port are set, create a http url from them
- otherwise, if UDS is available, use that
- lastly, if all else fails, default to localhost:8126

Reply:
```python
url = os.environ.get("DD_TRACE_AGENT_URL")
if not url:
    if HOST_SET or PORT_SET:
        url = "http://%s:%s" % (get_hostname(), get_trace_port())
    elif DEFAULT_UNIX_SOCKET_DOMAIN_AVAIL:
        url = "unix://%s" % (DEFAULT_UNIX_SOCKET_DOMAIN_PATH, )
    else:
        url = DEFAULT_TRACE_URL
```